api/millenium_patron.py

```diff
@@ -99,6 +106,31 @@ class MilleniumPatronAPI(BasicAuthenticationProvider, XMLParser):
           "default": PIN_AUTHENTICATION_MODE
         }
     ] + BasicAuthenticationProvider.SETTINGS
+
+    LIBRARY_SETTINGS = [
+        { "key": PATRON_FIELD_RESTRICTION_FIELD,
+          "label": _("Patron record restriction field"),
+          "type": "select",
+          "description": _("Field on the patron record used for <i>patron record restriction</i> below."),
+          "options": [
+              { "key": AGENCY_FIELD, "label": _(AGENCY_FIELD) },
+              { "key": PATRON_AGENCY_FIELD, "label": _(PATRON_AGENCY_FIELD) },
+              { "key": PATRON_TYPE_FIELD, "label": _(PATRON_TYPE_FIELD) },
+          ],
+          "default": PATRON_AGENCY_FIELD
+        },
+        { "key": PATRON_FIELD_RESTRICTION_STRING,
+          "label": _("Patron record restriction"),
+          "description": _("<p>When multiple libraries share an ILS, a person may be able to " +
+                           "authenticate with the ILS but not be considered a patron of " +
+                           "<i>this</i> library. This setting contains the rule for determining " +
+                           "whether an identifier is valid for this specific library.</p>" +
+                           "This compares the <i>patron record restriction field</i> on the patron " +
+                           "record to this text. Match is treated as a regular expression if it starts " +
+                           "with ^, otherwise a string match is used."),
```

Comment:
I would also mention that this is mutually exclusive with the patron identifier prefix setting. I'm not sure what would happen if you put in both but I'm guessing it wouldn't work.

Reply:
I don't think you would ever want to, but they do work fine together. The patron identifier prefix is tested first [here](https://github.com/NYPL-Simplified/circulation/blob/master/api/authenticator.py#L1455-L1485), then if it passes then this check will be done.

Reply:
Cool, that's fine then. You might still want to mention that you probably don't need that setting if you use this one, but the description's long already. Up to you :)